### DISCUSSION TITLE
How to add parent model types to the admin?

### DISCUSSION THREAD

**Discussion by mrmachine** [jazzband/django-polymorphic#710] [Q&A]
I might be missing something, but it seems like the parent model cannot be created and used directly in the admin. You MUST have at least one child model and you can ONLY create child models in the admin?

What if your parent model is a completely functional base type, and child models are _additional_ and _optional_? I don't want to create a concrete child subclass _just_ so I can create and edit it in the admin, when it does not require any additional fields.

The very first thing I see in the docs is an example of this, where `Project` appears to be functional on its own, with `ArtProject` and `ResearchProject` as subclasses. But these models are not used in the admin integration docs, and the example given uses generic `ModelA`, `ModelB` and `ModelC` where only `ModelB` and `ModelC` can be created and edited in the admin?

**Comment by merwok** [comment]
I was blocked by this issue too.  Adding an instance jumps to the ArtProject add form (if there is only one child model) or the subclass selection screen (if I have ArtProject and ResearchProject).  I had to define a proxy model for my parent model and put the proxy in child_models.

**Comment by mrmachine** [comment]
Sounds like a note in the docs explaining this limitation might be enough to resolve this issue, along with a recommendation to add a proxy model for the parent to `child_models` as a work-around, if allowing the parent to be used directly on its own is too hard?

**Comment by merwok** [comment]
I would wait for feedback from the maintainers here, so that they can say if this is a bug in the admin (since the examples show using the model directly is possible, as it would be without polymorphic) that can be fixed, or an expected limitation that can’t be avoided for technical reasons.

**Comment by bckohan** [comment]
Verified that this is in fact a bug. A few things going on - the base model is not included in the type selection - but also childless polymorphic models should also work, but those return a 404

**Comment by bckohan** [comment]
@mrmachine This does work - you just need to include the parent model type in the [child_models](https://django-polymorphic.readthedocs.io/en/stable/api/polymorphic.admin.html#polymorphic.admin.PolymorphicParentModelAdmin.child_models) tuple:

```python
class NoChildren(PolymorphicModel):
   ...


@register(NoChildren)
class NoChildrenAdmin(PolymorphicParentModelAdmin):
    child_models = (NoChildren,)

```
